### PR TITLE
chore(cli): @fern-api/openapi-parser is browser compatible

### PR DESCRIPTION
**Pull request by amckinney**
This updates the `@fern-api/openapi-parser` so that it is not browser compatible. This involved the following changes:

1. Update the `parse` method to accept a new `Document` interface (either OpenAPI or AsyncAPI).
2. Export additional common utilities from `@fern-api/openapi-parser` so that they can be used by `@fern-api/lazy-fern-workspace`.
3. Move the filesystem loader implementation to the `@fern-api/lazy-fern-workspace` package.
    * The implementation now exists in a new `OpenAPILoader` class.
4. Add the `InMemoryOpenAPILoader` interface to the `@fern-api/openapi-parser`, which can be safely used by browsers.
5. Add additional cases to `@fern-api/openapi-ir-to-fern-tests` to confirm that OpenAPI files parsed from a file can be used directly with the the `InMemoryOpenAPILoader`.

### CODE REVIEW

File: packages/cli/api-importers/openapi/openapi-ir-parser/src/index.ts [hunk 1]
```diff
@@ -1,2 +1,5 @@
 export { type ParseOpenAPIOptions } from "./options";
-export { parse, type Spec } from "./parse";
+export { parse, type Document } from "./parse";
+export { InMemoryOpenAPILoader } from "./openapi/InMemoryOpenAPILoader";
```
Comment (dsinghvi) [inline packages/cli/api-importers/openapi/openapi-ir-parser/src/index.ts:3]: i could see this being moved to the `browser-workspace` that you end up creating

**Review by dsinghvi — APPROVED** [review]

File: packages/cli/api-importers/openapi/openapi-ir-parser/src/openapi/v3/extensions/fernExtensions.ts [hunk 2]
```diff
@@ -1,5 +1,36 @@
...
+                    ...types.Operation,
+                    properties: {
+                        ...types.Operation?.properties,
+                        "x-fern-streaming": "XFernStreaming"
+                    }
+                }
+            };
+        }
+    }
+};
```
[the remark this answers is not in the document]
Comment (dsinghvi) [reply packages/cli/api-importers/openapi/openapi-ir-parser/src/openapi/v3/extensions/fernExtensions.ts:33]: i go both ways on this one (bc every loader will need to know to bring this in)

File: packages/cli/api-importers/openapi/openapi-ir-parser/src/index.ts (hunk 1 above)
Comment (amckinney) [reply packages/cli/api-importers/openapi/openapi-ir-parser/src/index.ts:3]: Great call - I have that tee'd up in the next PR 👍

File: packages/cli/api-importers/openapi/openapi-ir-parser/src/openapi/v3/extensions/fernExtensions.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (amckinney) [reply packages/cli/api-importers/openapi/openapi-ir-parser/src/openapi/v3/extensions/fernExtensions.ts:33]: Discussed offline - we're going to move the OpenAPI bundle options into the new `browser-workspace` package.

**Pull request merged**